packages/legend-query-builder/src/components/__tests__/QueryBuilderProjectVaraints.test.tsx

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Entity } from '@finos/legend-storage';
+import { describe, test, expect } from '@jest/globals';
+import {
+  act,
+  getAllByText,
+  getByText,
+  queryByText,
+  waitFor,
+} from '@testing-library/react';
+import TEST_DATA__QueryBuilder_Model_SimpleRelational from '../../stores/__tests__/TEST_DATA__QueryBuilder_Model_SimpleRelational.json';
+import {
+  TEST_DATA__projectWithColFunctionCollection,
+  TEST_DATA__projectWithColumnNameNotCollection,
+  TEST_DATA__projectWithLambdaNotCollection,
+  TEST_DATA__projectWithSingleColFunction,
+  TEST_DATA__simpleBasicProject,
+} from '../../stores/__tests__/TEST_DATA__QueryBuilder_ProjectVaraints.js';
+import { integrationTest } from '@finos/legend-shared/test';
+import { TEST__setUpQueryBuilder } from '../__test-utils__/QueryBuilderComponentTestUtils.js';
+import {
+  create_RawLambda,
+  extractElementNameFromPath,
+  stub_RawLambda,
+} from '@finos/legend-graph';
+import { TEST_DATA__ModelCoverageAnalysisResult_SimpleRelationalWithExists } from '../../stores/__tests__/TEST_DATA__ModelCoverageAnalysisResult.js';
+import { QUERY_BUILDER_TEST_ID } from '../../__lib__/QueryBuilderTesting.js';
+import { guaranteeType } from '@finos/legend-shared';
+import { QueryBuilderTDSState } from '../../stores/fetch-structure/tds/QueryBuilderTDSState.js';
+
+type TestCase = [
```

Comment:
can we test the actual writing of these projection lambdas. 
So for these, we should be able to read them but still write to our default project variant. Can we test this writing of the rawlambda from the UI state that has been built?

Reply:
A `expectedRawLambda` has been added to check whether the rawLambda built is expected